### DISCUSSION TITLE
Ever a possibility for a model that attempts to correct or replace Text written by Stable Diffusion that is jumbled?

### DISCUSSION THREAD

**Discussion by CCpt5** [Sanster/IOPaint#221] [Ideas]
Hello,

I know this is probably not possible at this time, but I wanted to mention it just in case. It's well known that Stable Diffusion (and most of the AI txt to image tools) don't properly generate text. They put letters/numbers/symbols where text should be, but it's usually jumbled/etc.  
  
I use Lama a lot to remove text from images SD generates that end up w/ text in them. Removing the text is pretty easy w/ Lama, and doing so makes images that had text look less obviously AI generated. That said, it'd be amazing if in the future a model trained on letters/fonts/etc might be able to recreate/replace text that SD has incorporated into an image and also maintain a similar style to how it appears on the image already (which is usually aesthetically correct).

Just thinking out loud. If anyone has thoughts on how likely this functionality/type of model may be in time I'd love to hear.

Thx.

**Comment by CCpt5** [comment]
Here are a couple examples of images that I generated today w/ SD that I would love to be able to modify the non-sensical text on while keeping the font/style. 

![00047-3593841265-topnotch artstyle](https://user-images.githubusercontent.com/37972078/219937003-b057d258-7fe0-4561-9171-8488a3f6e4cd.png)  ![04046-2075483986-topnotch trending on artstation, 8 k](https://user-images.githubusercontent.com/37972078/219937116-1eecbd3a-c4f7-43b8-889e-20b44801f5b2.png)

**Comment by Sanster** [comment]
This is a very interesting idea for ai-assisted creation, after a quick search I found this project https://prasunroy.github.io/stefann/

![90DBA078-3F24-4E45-99AE-4B903EE7BD4E](https://user-images.githubusercontent.com/3998421/219983648-354de8ad-8f3b-499e-b227-e1def917ef51.png)
![691A2F04-AA91-4994-ADB3-8B76165C933C](https://user-images.githubusercontent.com/3998421/219983655-3b3f4d8f-6227-401f-85aa-e2e07e9c6680.png)

**Reply by CCpt5** [reply]
Wow! That's exactly what I'm hoping for.

Is there any chance this could be incorporated into a future version of Lama?

**Reply by Sanster** [reply]
Did a quick test on your image, I would say the result is not good, and actually the 'replace process' is not easy(there are many algorithm parameters that need to be adjusted). 

<img width="1016" alt="截屏2023-02-20 16 49 54" src="https://user-images.githubusercontent.com/3998421/220222968-952fa351-de1c-4527-868d-b5d2a9433123.png">

It would definitely be a good feature to have in Lama Cleaner if there is a model that works better and is easier to use. Let's wait and see.
